### PR TITLE
copy: add fast copy path for darwin

### PR DESCRIPTION
**Pull request by tonistiigi**
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

**Comment by tiborvass** [comment]
Should you have an `_other.go` file with `copyFile` to avoid compile errors on other platforms for code importing this package?

**Comment by tonistiigi** [comment]
It already exists for linux/windows. We don't have any way to test others anyway.

File: copy/copy_linux.go [hunk 1]
```diff
@@ -52,6 +52,21 @@ func (c *copier) copyFileInfo(fi os.FileInfo, name string) error {
 	return nil
 }
 
+func copyFile(source, target string) error {
```
Comment (tiborvass) [inline copy/copy_linux.go]: you could call this `copyFileNaive` in `copy.go` and call it as a fallback from darwin.

Comment (tonistiigi) [reply copy/copy_linux.go]: But this function is not naive in linux. It does copy_file_range on linux.

**Pull request merged**